docs/guide.rst

```diff
@@ -86,3 +86,141 @@ options::
+
+:ref:`/spec/plans` are used to enable testing and group relevant
+tests together. They describe how to :ref:`/spec/plans/discover`
+tests for execution, how to :ref:`/spec/plans/provision` the
+environment, how to :ref:`/spec/plans/prepare` it for testing, how
+to :ref:`/spec/plans/execute` tests, :ref:`/spec/plans/report`
+results and finally how to :ref:`/spec/plans/finish` the test job.
+
+Here's an example of a slightly more complex plan which changes
+the default provision method to container to speed up the testing
+process and ensures that an additional package is installed before
+the testing starts::
+
+    provision:
+        how: container
+        image: fedora:33
+    prepare:
+        how: install
+        package: wget
+    execute:
+        how: tmt
+        script: wget http://example.org/
+
+
+Tests
+------------------------------------------------------------------
+
+Very often testing is much more complex than running just a
+single shell script. There might be many scenarios covered by
+individual scripts. For these cases the ``discover`` step can
+be instructed to explore available tests from fmf metadata as
+well. The plan will look like this::
+
+    discover:
+        how: fmf
+    execute:
+        how: tmt
+
+Tests, identified by the required key ``test``, define attributes
+which are closely related to individual test cases such as the
+:ref:`/spec/tests/test` script, :ref:`/spec/tests/framework`,
+directory :ref:`/spec/tests/path` where the test should be
+executed, maximum test :ref:`/spec/tests/duration` or packages
+required to run the test. Here's an example of test metadata::
+
+    summary: Fetch an example web page
+    test: wget http://example.org/
+    require: wget
+    duration: 1m
+
+Similar to plans, it is possible to choose an arbitrary name for
+the test. Just make sure the ``test`` key is defined. However, to
+organize the metadata efficiently it is recommended to keep tests
+and plans under separate folders, e.g. ``tests`` and ``plans``.
+This will also allow you to use `inheritance`__ to prevent
+unnecessary data duplication.
+
+__ https://fmf.readthedocs.io/en/latest/features.html#inheritance
+
+
+Stories
+------------------------------------------------------------------
+
+It's always good to start with a "why". Or, even better, with a
+story which can describe more context behind the motivation.
+:ref:`/spec/stories` can be used to track implementation, test and
+documentation coverage for individual features or requirements.
+Thanks to this you can track everything in one place, including
+the project implementation progress. Stories are identified by the
+``story`` attribute which every story has to define or inherit.
+
+An example story can look like this::
+
+    story:
+        As a user I want to see more detailed information for
+        particular command.
+    example:
+      - tmt test show -v
+      - tmt test show -vvv
+      - tmt test show --verbose
+
+
+Core
```

Comment:
Not sure how I feel about having Core at the end vs in the beginning. Both make sense to me in a way, they are optional, although nice to have and not essential to the explanation of plans/tests/stories... On the other hand it's nice to know from the beginning that one should include a summary :)

Reply:
Yeah, the `Core` description would logically belong to the top. However, after a bit of thinking, I decided to start the chapter with plans which are more tangible (e.g. the minimal plan). Keeping these common attributes at the very end then looks like the best remaining option. I don't think the extra unexplained `summary` key will confuse anybody.